File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/2112-minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        if len(nums) <= 1:
            return 0
        res = float("inf")
        nums.sort()
        l = 0
        for r in range(k - 1 , len(nums)):
            res = min(res , nums[r] - nums[l])
            l += 1
        return res

File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/2112-minimum-difference-between-highest-and-lowest-of-k-scores.java
class Solution {
    public int minimumDifference(int[] nums, int k) {
        if(nums.length <= 1){
            return 0;
        }
        int res =Integer.MAX_VALUE; 
        int l = 0;
        Arrays.sort(nums);
        for(int r = k - 1 ; r < nums.length ; r++){
            res = Math.min(res , nums[r] - nums[l]);
            l += 1;
        }
        return res;
    }
}